File: EasyUpiPayment/src/main/java/com/shreyaspatil/EasyUpiPayment/model/PaymentApp.java
package com.shreyaspatil.EasyUpiPayment.model;

public enum PaymentApp {
    AMAZON_PAY(Package.AMAZON_PAY),
    BHIM_UPI(Package.BHIM_UPI),
    GOOGLE_PAY(Package.GOOGLE_PAY),
    PAYTM(Package.PAYTM),
    PHONE_PE(Package.PHONE_PE);

    private String mPackageName;

    PaymentApp(String mPackageName) {
        this.mPackageName = mPackageName;
    }

    public String getPackageName() {
        return mPackageName;
    }

    private static final class Package {
        static final String AMAZON_PAY = "in.amazon.mShop.android.shopping";
        static final String BHIM_UPI = "in.org.npci.upiapp";
        static final String GOOGLE_PAY = "com.google.android.apps.nbu.paisa.user";
        static final String PHONE_PE = "com.phonepe.app";
        static final String PAYTM = "net.one97.paytm";
    }
}

File: EasyUpiPayment/src/main/java/com/shreyaspatil/EasyUpiPayment/model/PaymentApp.kt
package com.shreyaspatil.easyupipayment.model

@Suppress("SpellCheckingInspection")
enum class PaymentApp(val packageName: String) {
	ALL(Package.ALL),
	AMAZON_PAY(Package.AMAZON_PAY),
	BHIM_UPI(Package.BHIM_UPI),
	GOOGLE_PAY(Package.GOOGLE_PAY),
	PAYTM(Package.PAYTM),
	PHONE_PE(Package.PHONE_PE);

	private object Package {
		const val ALL = "ALL"
		const val AMAZON_PAY = "in.amazon.mShop.android.shopping"
		const val BHIM_UPI = "in.org.npci.upiapp"
		const val GOOGLE_PAY = "com.google.android.apps.nbu.paisa.user"
		const val PHONE_PE = "com.phonepe.app"
		const val PAYTM = "net.one97.paytm"
	}
}